packages/core/services/slack/constructBirthdaySquadWelcomeMessage.ts

```diff
@@ -0,0 +1,99 @@
+import { getIceBreakerThreads } from "@/db/queries/getIceBreakerThreads";
+import { getPresentIdeasByUser } from "@/db/queries/getPresentIdeasByUser";
+import type { PostMessageArguments } from "@/types/MessageArguments";
+
+import { getIceBreakerThreadLink } from "./getIceBreakerThreadLink";
+
+type Arguments = {
+  channel: string;
+  teamId: string;
+  birthdayPerson: string;
+  eventId?: string;
+};
+
+export const welcomeMessageEventType = "sendSquadWelcomeMessage";
+
+const previousPresentsSheet =
+  "https://docs.google.com/spreadsheets/d/1pzdNkF-18OUS6qhcVD2HHGkTwTmCXJEvR8LBzPjUnGI/edit?pli=1#gid=0";
+
+const formatIceBreakerLinks = (iceBreakerLinks: string[]) =>
+  iceBreakerLinks.map((iceBreakerLink) => `• ${iceBreakerLink}`).join("\n");
+
+const formatPresentIdeas = (presentIdeas: string[]) =>
+  presentIdeas
+    .map((idea) =>
+      idea
+        .split("\n")
+        .map((line) => `> ${line}`)
+        .join("\n"),
+    )
+    .join("\n\n");
+
+export const constructBirthdaySquadWelcomeMessage = async ({
+  channel,
+  teamId,
+  birthdayPerson,
+  eventId,
+}: Arguments): Promise<PostMessageArguments> => {
+  const iceBreakers = await getIceBreakerThreads(teamId, birthdayPerson);
+
+  const iceBreakerLinkResponses = await Promise.all(
+    iceBreakers.map(getIceBreakerThreadLink),
+  );
+
+  const iceBreakerLinks = formatIceBreakerLinks(
+    iceBreakerLinkResponses.flatMap(
+      (iceBreakerLink) => iceBreakerLink.permalink ?? [],
+    ),
+  );
+
+  const presentIdeas = await getPresentIdeasByUser(teamId, birthdayPerson);
+
+  const presentIdeasText = Array.from(presentIdeas.entries())
+    .map(([user, ideas]) => `<@${user}>:\n${formatPresentIdeas(ideas)}\n`)
+    .join("\n");
+
+  return {
+    channel,
+    text: `Welcome to the birthday squad of <@${birthdayPerson}>! 🎁`,
+    blocks: [
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: `*Welcome to the birthday squad of <@${birthdayPerson}>! 🎁*`,
+        },
+      },
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: `*Icebreaker threads*\n${iceBreakerLinks}`,
+        },
+      },
+      {
+        type: "section",
```

Comment:
Should we skip this section if there is no presentIdea?

Reply:
It shouldn't happen tbh. 😄 But I changed it.